src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -130,9 +127,19 @@ export async function fetchGenericAssayData(
             } as GenericAssayDataFilter,
         };
     });
-    const dataPromises = params.map(param =>
-        client.fetchGenericAssayDataInMolecularProfileUsingPOST(param)
-    );
+    const dataPromises = params.map((param: any) => {
+        // do not request data by using empty sample list
+        if (
+            _.isEmpty(param.genericAssayDataFilter.sampleIds) &&
+            !param.genericAssayDataFilter.sampleListId
+        ) {
+            return Promise.resolve([]);
+        } else {
+            return client.fetchGenericAssayDataInMolecularProfileUsingPOST(
+                param
+            );
+        }
+    });
```

Comment:
This is the fix for issue: cBioPortal/cbioportal#7793. If we are using custom sample list for multiple study query, we can get a query with empty sample list, we need to skip data fetching for this case.